rundeckapp/grails-app/services/rundeck/services/ReportService.groovy

```diff
@@ -302,11 +302,11 @@ class ReportService  {
                             or {
                                 List execIdFilterPartioned = Lists.partition(query.execIdFilter, 1000)
                                 execIdFilterPartioned.each { List partition ->
-                                    'in'('jcExecId', partition)
+                                    'in'('executionId', partition.collect{Long.parseLong(it)})
```

Comment:
To further improve the performance we could avoid the use of "in" sentence by using DetachedCriteria, and instead of receive the execution id list we could just receive the authorized projects list (that should be fairly smaller) to restrict the join and then attach the subquery with exist statement

Something like this:

``` 
subquery = DetachedCriteria.forClass(ReferencedExecution.class, "re").with{
                setProjection Projections.property('re.execution.id')
                add Restrictions.conjunction().add(Restrictions.eq('re.executionId', 'this_.jcExecId'))
                add Restrictions.conjunction().add(Restrictions.eq('re.scheduledExecution.id', se.id))
                if(query.execProjects){
                    add Restrictions.conjunction().add(Restrictions.in('this.ctxProject', query.execProjects))
                }
            }
```